src/components/PatientProfilePage/PatientProfilePage.tsx

```diff
@@ -95,39 +141,95 @@ const PatientProfilePage = ({
         status,
         runNumber,
       });
+      if (hospitalId) setSelectedHospital(hospitalId);
+      if (ambulanceId) setSelectedAmbulance(ambulanceId);
+      setTransportConfirmed(status === Status.TRANSPORTED);
     }
-  }, [data]);
+  }, [data, loading, mode]);
+
   useEffect(() => {
     if (mode === 'new' && barcodeValue) {
       setFormFields({ ...formFields, barcodeValue });
     }
-  }, []);
+  }, [mode, barcodeValue]);
 
-  const [addPatient] = useMutation(ADD_PATIENT, {
-    update(cache, { data: { newPatient } }) {
-      cache.writeQuery({
-        query: GET_ALL_PATIENTS,
-        data: { patients: patients.concat([newPatient]) },
-      });
-    },
-  });
-  const [editPatient] = useMutation(EDIT_PATIENT);
+  const handleCloseDialog = () => {
+    setOpenTransportModal(false);
+  };
+
+  const handleCancelTransport = () => {
+    setSelectedHospital({
+      id: '',
+      name: '',
+    });
+    setSelectedAmbulance({
+      id: '',
+      vehicleNumber: 0,
+    });
+    setOpenTransportModal(false);
+    setOpenTransportPage(false);
+  };
+
+  const handleConfirmTransport = () => {
+    setTransportConfirmed(true);
+    setOpenTransportPage(false);
+  };
+
+  const handleHospitalChange = (e: React.ChangeEvent<HTMLElement>) => {
+    setSelectedHospital({
+      id: (e.target as HTMLInputElement).value,
+      name: hospitals.filter(
+        (hospital) => hospital.id === (e.target as HTMLInputElement).value
+      )[0].name,
+    });
+  };
+
+  const handleAmbulanceChange = (e: React.ChangeEvent<HTMLElement>) => {
+    setSelectedAmbulance({
+      id: (e.target as HTMLInputElement).value,
+      vehicleNumber: ambulances.filter(
+        (ambulance) => ambulance.id === (e.target as HTMLInputElement).value
+      )[0].vehicleNumber,
+    });
+  };
+
+  const action = () => (
+    <>
+      <Button style={{ color: Colours.SnackbarButtonBlue }}>
+        View Patient Details
+      </Button>
+    </>
+  );
 
   const handleComplete = () => {
+    if (transportingPatient && !transportConfirmed) {
+      setOpenTransportModal(true);
+      return;
+    }
+    let transportTime;
+    if (transportingPatient && transportConfirmed) {
+      transportTime = new Date();
+    }
```

Comment:
minor refactoring but also brings up another question - do we want the client to be responsible for updating this timestamp? the db maintains the `updatedAt` and `createdAt` fields so it would potentially make sense to not handle times here - if we like this idea it can be part of a different ticket

cc: @wongcmichelle for your thoughts
```suggestion
    const transportTime = transportingPatient && transportConfirmed ? new Date() : null;
```

Reply:
If `transportTime` were ever edited anywhere else in the FE, I would say handling it in the backend would be ideal/cleaner. However, since that doesn't seem to be the case, I'm okay with it either way.

Reply:
That is to say, if a patient is ever "untransported" we should be sure to handle that too.

Reply:
i'll create a ticket regardless - in case untransporting is possible; if it is and we need to implement it, we can reconsider where this is handled